toolkit/block-production-log/pallet/src/lib.rs

```diff
@@ -92,12 +101,23 @@ mod test;
 pub use pallet::*;
 pub use weights::WeightInfo;
 
+/// Source of the current block's author
+pub trait GetAuthor<Moment, BlockProducerId> {
+	/// Returns the current block's author
+	fn get_author(moment: &Moment) -> BlockProducerId;
+}
```

Comment:
can we use https://docs.rs/frame-support/latest/frame_support/traits/trait.FindAuthor.html for that? Usually it is implemented as mapping from PreDigest of specific consensus engine to author index in committee:
https://github.com/paritytech/polkadot-sdk/blob/5ab401310d2fbd808b3ffbcfc8735ad8f1c8fe10/substrate/frame/aura/src/lib.rs#L348